### PR TITLE
Add filters for status_widget_stock_rows queries

### PR DESCRIPTION
**Pull request by ghost**
…d_Products widget

### All Submissions:

* [x] Have you followed the [WooCommerce Contributing guideline](https://github.com/woocommerce/woocommerce/blob/master/.github/CONTRIBUTING.md)?
* [x] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../../pulls) for the same update/change?

### Changes proposed in this Pull Request:

Add the possibility to filter status_widget_stock_rows queries.

### Other information:

* [ ] Have you added an explanation of what your changes do and why you'd like us to include them?
* [ ] Have you written new tests for your changes, as applicable?
* [x] Have you successfully run tests with your changes locally?

### Changelog entry

> Dev - Added pre query filters for WP dashboard WooCommerce status widget.

### CODE REVIEW

File: includes/admin/class-wc-admin-dashboard.php [hunk 1]
```diff
@@ -214,7 +214,9 @@ private function status_widget_stock_rows() {
 			$lowinstock_count = get_transient( $transient_name );
 
 			if ( false === $lowinstock_count ) {
-				$lowinstock_count = (int) $wpdb->get_var(
+
+				$lowinstock_count_query = apply_filters(
+					'woocommerce_status_widget_lowinstock_count_query',
```
Comment (claudiosanches) [inline includes/admin/class-wc-admin-dashboard.php]: Please change to `woocommerce_status_widget_low_in_stock_count_query`.

File: includes/admin/class-wc-admin-dashboard.php [hunk 2]
```diff
@@ -226,14 +228,17 @@ private function status_widget_stock_rows() {
...
 			}
 
 			$transient_name   = 'wc_outofstock_count';
 			$outofstock_count = get_transient( $transient_name );
 
 			if ( false === $outofstock_count ) {
-				$outofstock_count = (int) $wpdb->get_var(
+
+				$outofstock_count_query = apply_filters(
+					'woocommerce_status_widget_outofstock_count_query',
```
Comment (claudiosanches) [inline includes/admin/class-wc-admin-dashboard.php]: Please update to `woocommerce_status_widget_out_of_stock_count_query`.

**Review by claudiosanches — CHANGES_REQUESTED** [review]
Looks good in general, just two small tweaks related to filter names.
Please let me know if you are able to update it.
Thanks.

File: includes/admin/class-wc-admin-dashboard.php (hunk 1 above)
Comment (grey-rsi) [reply includes/admin/class-wc-admin-dashboard.php]: @claudiosanches changed, Thanks!

File: includes/admin/class-wc-admin-dashboard.php (hunk 2 above)
Comment (grey-rsi) [reply includes/admin/class-wc-admin-dashboard.php]: @claudiosanches changed, Thanks!

**Review by claudiosanches — APPROVED** [review]
Thanks for your contribution!

**Review by claudiosanches — APPROVED** [review]
Looks good, and it's similar with what we do for queries of stock reports, so makes sense to include here too.

**Comment by grey-rsi** [comment]
@claudiosanches Thanks for merging this!

**Review by claudiosanches — CHANGES_REQUESTED** [review]
Hello, thank you for your help.

I discussed this PR with some members of my team and I think we should change it to a "pre query filter" instead of allowing replacing the query.

See an example:

```php
			if ( false === $lowinstock_count ) {
				/**
				 * Status widget low in stock count pre query.
				 *
				 * @since 5.3
				 * @param null|string $low_in_stock_count Low in stock count, by default null.
				 * @param int         $stock              Low stock amount.
				 * @param int         $nostock            No stock amount
				 */
				$lowinstock_count = apply_filters( 'woocommerce_status_widget_low_in_stock_count_pre_query', null, $stock, $nostock );

				if ( ! is_null( $lowinstock_count ) ) {
					$lowinstock_count = (int) $wpdb->get_var(
						$wpdb->prepare(
							"SELECT COUNT( product_id )
							FROM {$wpdb->wc_product_meta_lookup} AS lookup
							INNER JOIN {$wpdb->posts} as posts ON lookup.product_id = posts.ID
							WHERE stock_quantity <= %d
							AND stock_quantity > %d
							AND posts.post_status = 'publish'",
							$stock,
							$nostock
						)
					);
				}
				set_transient( $transient_name, $lowinstock_count, DAY_IN_SECONDS * 30 );
			}
```

So we avoid any incompatibility in the future.

Let me know if you are willing to perform those updates, if not I can try write a PR for it and you can help review.
Thank you again!

**Comment by grey-rsi** [comment]
@claudiosanches changed to pre-filter. Please review Thanks!

**Comment by peterfabian** [comment]
Hi, maybe I'm misunderstanding the purpose of these filters, but doesn't it mean that by default those queries would be skipped?

**Comment by grey-rsi** [comment]
@peterfabian you're right, thanks! There was a typo in the example which I have copied in the code, sorry for this...

**Review by claudiosanches — APPROVED** [review]
Thank you very much for all your help!
Now with pre-query filters looks much better.

**Pull request merged**